src/encord_active/lib/common/project.py

```diff
@@ -0,0 +1,245 @@
+        for lr_hash in self.label_row_meta.keys():
+            lr_file_path = project_dir / "data" / lr_hash / "label_row.json"
+            lr_images_dir = project_dir / "data" / lr_hash / "images"
+            if not lr_file_path.exists():  # todo log this issue
+                continue
+            if not lr_images_dir.exists() or not lr_images_dir.is_dir():  # todo log this issue
+                continue
+            self.label_rows[lr_hash] = LabelRow(json.loads(lr_file_path.read_text()))
+            self.image_paths[lr_hash] = list(lr_images_dir.iterdir())
+
+    @classmethod
+    def from_encord_project(cls, project_dir: Path, encord_project: EncordProject):
+        """
+        Construct a Project from Encord platform.
+
+        :param project_dir: The root directory of the project.
+        :param encord_project: Encord project from where the data is fetched.
+        :return:
+        """
+        if not project_dir.exists():
+            raise FileNotFoundError("`project_dir` must point to an existing directory")
+        if not project_dir.is_dir():
+            raise NotADirectoryError("`project_dir` must point to an existing directory")
+
+        # todo enforce clean up when we are sure it won't impact performance in other sections (like PredictionWriter)
+        # also don't forget to add `from shutil import rmtree` at the top (pylint tags it as unused right now)
+        # clean project_dir content
+        # for path in project_dir.iterdir():
+        #     if path.is_file():
+        #         path.unlink()
+        #     elif path.is_dir():
+        #         rmtree(path)
+
+        # store project's metadata
+        project_meta = {
+            "project_title": encord_project.title,
+            "project_description": encord_project.description,
+            "project_hash": encord_project.project_hash,
+        }
+        project_meta_file_path = project_dir / "project_meta.yaml"
+        project_meta_file_path.write_text(yaml.dump(project_meta))  # , encoding="utf-8")
+
+        # store project's ontology
+        ontology_file_path = project_dir / "ontology.json"
+        ontology_file_path.write_text(json.dumps(encord_project.ontology, indent=2))  # , encoding="utf-8")
+
+        # store label rows' metadata
+        label_row_meta = {lr["label_hash"]: lr for lr in encord_project.label_rows if lr["label_hash"] is not None}
+        label_row_meta_file_path = project_dir / "label_row_meta.json"
+        label_row_meta_file_path.write_text(json.dumps(label_row_meta, indent=2))  # , encoding="utf-8")
+
+        # store label rows and their images
+        label_rows = download_all_label_rows(encord_project, cache_dir=project_dir)  # todo no need to output all data
+        image_paths = download_all_images(label_rows, cache_dir=project_dir)  # todo no need to output all data
+
+        return Project(project_dir)
+
+    # unused method (prototype)
+    def save(self, cache_dir: Path) -> None:
```

Comment:
I think this is kind of useless at this point since we don't really change the local project and this isn't used by the `from_encord_project` which seems to replicate this logic.

I don't want to delay this PR event more, but I think we should iterate over this in the future.

I think that separating what's in memory and what's on disc can be useful here. We should be able to create a `Project` from an encord project without saving anything and only when `save` is called, we dump everything to disc. 

Also, in the future, we might want the ability to have a flow of creating a project from the UI and for that, we will also need to be able to instantiate a `Project` with values rather than a path. (loading from disc would be in a `load` method for example)

Reply:
I like the idea of separating memory and disk stuff but it will only work for prototyping, not production use cases as you would have to handle a massive amount of images/videos or to always have an open connection with the server that provides the data which adds way more time overload.

Reply:
Sorry, I wasn't clear, I don't think we should be holding all images in memory, just the data. We can fetch the media on demand or require an explicit download before media can be displayed, but the interim in-memory object could still be valuable.